src/Features/Core/Portable/CodeLens/ReferenceLocationDescriptor.cs

```diff
@@ -9,14 +9,35 @@ namespace Microsoft.CodeAnalysis.CodeLens
     /// </summary>
     internal sealed class ReferenceLocationDescriptor
     {
-        public int LineNumber { get; }
-
-        public int ColumnNumber { get; }
-
         public Guid ProjectGuid { get; }
 
         public Guid DocumentGuid { get; }
 
+        /// <summary>
+        /// Document's file path
+        /// </summary>
+        public string FilePath { get; }
+
+        /// <summary>
+        /// Reference's Span Start
+        /// </summary>
+        public int SpanStart { get; }
```

Comment:
so, this contains 2 different info. one is against document and one is against a line from the document.

so, filepath, project/documentGuid, spanStart, spanLength, LineNumber, ColumnNumber is all against Document.

ReferenceStart and ReferenceLength is against ReferenceLine.

Reply:
> One is against document and one is against a line from the document.

Why do we need both?  Isn't the document version enough info?

Reply:
might be. but not planning to clean that up in this PR. one thing for sure is existing info is not enough since existing info only contains processed info but not original info (original spans)

Reply:
➡️ #31061